extensions/amp-social-share/0.2/social-share.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Keys} from '../../../src/utils/key-codes';
+import {Services} from '../../../src/services';
+import {addParamsToUrl, parseQueryString} from '../../../src/url';
+import {createElement, useContext, useState} from '../../../src/preact';
+import {dev, devAssert, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getAmpContext} from '../../../src/preact/context';
+import {getSocialConfig} from './amp-social-share-config';
+import {openWindowDialog} from '../../../src/dom';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+const socialShareStyle = `background-repeat: no-repeat;
```

Comment:
Can't we import this from the css? Like https://github.com/ampproject/amphtml/blob/6d7ccee0e30987d546459d170d24c49d9419bddb/extensions/amp-social-share/0.1/amp-social-share.js#L17?

Reply:
This style declaration is temporary while we figure out a generic solution for handling CSS in Bento. 

The build outputs all of the styles in that file as a single string assigned to the `CSS` const. It doesn't work to append this as is, since `style="selector={}..."` for instance is meaningless inline. One approach would be to import this, parse the string into a map of `selector` to string `declaration block`, and append styles based on usage `type`. This is probably ill-advised for runtime performance, so a better solution might be to modify our build to output the CSS into a different shape.

Still, none of these sound like an elegant solution that we'll want to go with long term, so I'd rather do the thing that works most cheaply in the short term.